src/main/java/com/example/contactsservice/Service/ContactService.java

```diff
@@ -0,0 +1,68 @@
+package com.example.contactsservice.Service;
+
+import com.example.contactsservice.BadRequest;
+import com.example.contactsservice.DTO.ContactDTO;
+import com.example.contactsservice.Entity.ContactEntity;
+import com.example.contactsservice.NotFound;
+import com.example.contactsservice.Repository.ContactRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ContactService {
+    @Autowired
+    ContactRepository contactRepository;
+
+    public ContactEntity createContact(ContactDTO contactDTO) {
+        validateContactDto(contactDTO);
+        ContactEntity contact = getContact(contactDTO, new ContactEntity());
+        return contactRepository.save(contact);
+    }
+
+    public ContactEntity getContact(Long id) {
+        ContactEntity contact = contactRepository.getById(id);
+        try {
+            if (null != contact.getFirstName()) {
+                return contact;
+            }
+            return null;
+        } catch (RuntimeException runtimeException) {
```

Comment:
runtime error could be from different error scenarios. how do you know that it is always NotFound error?

Reply:
also this `try` block only has `contact.getFirstName` that is not going to raise any exception. In fact line `getById` method can throw some runtime error.